src/unity/lib/unity_sarray.cpp

```diff
@@ -234,66 +234,80 @@ void unity_sarray::construct_from_json_record_files(std::string url) {
+            buffer.resize(fin.file_size());
+            buffer.shrink_to_fit();
+            fin.read(buffer.data(), fsize);
+
+            // try to parse. failing on error
+            const char* str = buffer.data();
+            auto parse_result = parser.recursive_parse(&str, fsize);
+            if (parse_result.second == false ||
+                parse_result.first.get_type() != flex_type_enum::LIST) {
+              std::stringstream error_msg;
+              error_msg << "Unable to parse " << sanitize_url(p.first) << ". "
+                        << "It does not appear to be in JSON record format. "
+                        << "A list of dictionaries is expected" << std::endl;
+
+              log_and_throw(error_msg.str());
+            }
 
-        size_t num_elems_parsed = 0;
-        bool has_non_dict_elements = false;
-        for (const auto& element : parse_result.first.get<flex_list>()) {
-          if (element.get_type() == flex_type_enum::DICT ||
-              element.get_type() == flex_type_enum::UNDEFINED) {
-            (*output) = element;
-            ++output;
-            ++num_elems_parsed;
-          } else {
-            has_non_dict_elements = true;
-          }
-        }
+            size_t num_elems_parsed = 0;
+            bool has_non_dict_elements = false;
+            for (const auto& element : parse_result.first.get<flex_list>()) {
+              if (element.get_type() == flex_type_enum::DICT ||
+                  element.get_type() == flex_type_enum::UNDEFINED) {
+                (*output) = element;
+                ++output;
+                ++num_elems_parsed;
+              } else {
+                has_non_dict_elements = true;
+              }
+            }
 
-        logstream(LOG_PROGRESS) << "Successfully parsed an SArray of "
-                                << num_elems_parsed 
-                                << " elements from the JSON file "
-                                << sanitize_url(p.first);
+            logstream(LOG_PROGRESS)
+                << "Successfully parsed an SArray of " << num_elems_parsed
+                << " elements from the JSON file " << sanitize_url(p.first);
 
-        if (has_non_dict_elements) {
-          logstream(LOG_PROGRESS) << sanitize_url(p.first)
-                                  << " has non-dictionary elements which are ignored. " << std::endl;
+            if (has_non_dict_elements) {
+              logstream(LOG_PROGRESS)
+                  << sanitize_url(p.first)
+                  << " has non-dictionary elements which are ignored. "
+                  << std::endl;
+            }
+          } else {
+            logstream(LOG_PROGRESS)
+                << "Unable to read " << sanitize_url(p.first) << std::endl;
+          }
+          break;
         }
-      } else {
-        logstream(LOG_PROGRESS) << "Unable to read " << sanitize_url(p.first) << std::endl;
-      }
-    }
+      case fileio::file_status::DIRECTORY:
+        break; 
```

Comment:
Should reading from a directory error out with a user-facing message? It seems this logic may result in an empty SArray/SFrame when attempting to read a directory, which is not quite what the user might expect.

Reply:
Updated in PR